soh/src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c

```diff
@@ -487,12 +487,46 @@ void EnExItem_DrawItems(EnExItem* this, PlayState* play) {
     }
     if (this) {}
     func_8002ED80(&this->actor, play, 0);
+    if (IS_RANDO) {
+        GetItemEntry randoGetItem = (GetItemEntry)GET_ITEM_NONE;
+        switch (this->type) {
+            case EXITEM_BOMB_BAG_BOWLING:
+            case EXITEM_BOMB_BAG_COUNTER:
+                randoGetItem = (CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("MysteriousShuffle"), 0) && Randomizer_IsCheckShuffled(RC_MARKET_BOMBCHU_BOWLING_FIRST_PRIZE))
+                        ? GetItemMystery()
+                        : Randomizer_GetItemFromKnownCheck(RC_MARKET_BOMBCHU_BOWLING_FIRST_PRIZE, GI_BOMB_BAG_20);
+                break;
+            case EXITEM_BOMBCHUS_BOWLING:
+            case EXITEM_BOMBCHUS_COUNTER:
+                randoGetItem = Randomizer_GetItemFromKnownCheck(RC_MARKET_BOMBCHU_BOWLING_BOMBCHUS, GI_BOMBCHUS_10);
+                break;
+            case EXITEM_BULLET_BAG:
+                randoGetItem = (CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("MysteriousShuffle"), 0) && Randomizer_IsCheckShuffled(RC_LW_TARGET_IN_WOODS))
+                                ? GetItemMystery() : Randomizer_GetItemFromKnownCheck(RC_LW_TARGET_IN_WOODS, GI_BULLET_BAG_50);
+                break;
+        }
+
+        if (randoGetItem.getItemId != GI_NONE) {
+            EnItem00_CustomItemsParticles(&this->actor, play, randoGetItem);
+            GetItemEntry_Draw(play, randoGetItem);
+            return;
+        }
+    }
+
     GetItem_Draw(play, this->giDrawId);
 }
 
 void EnExItem_DrawHeartPiece(EnExItem* this, PlayState* play) {
     func_8002ED80(&this->actor, play, 0);
-    GetItem_Draw(play, GID_HEART_PIECE);
+
+    if (IS_RANDO) {
+        GetItemEntry randoGetItem = (CVarGetInteger(CVAR_RANDOMIZER_ENHANCEMENT("MysteriousShuffle"), 0) && Randomizer_IsCheckShuffled(RC_MARKET_BOMBCHU_BOWLING_SECOND_PRIZE))
+                                    ? GetItemMystery() : Randomizer_GetItemFromKnownCheck(RC_MARKET_BOMBCHU_BOWLING_SECOND_PRIZE, GI_HEART_PIECE);
+        EnItem00_CustomItemsParticles(&this->actor, play, randoGetItem);
+        GetItemEntry_Draw(play, randoGetItem);
+    } else {
+        GetItem_Draw(play, GID_HEART_PIECE);
+    }
```

Comment:
This has been moved to `EnExItem_DrawRandomizedItem
```suggestion
    GetItem_Draw(play, GID_HEART_PIECE);
```